code/modules/projectiles/guncrafting/frame.dm

```diff
@@ -0,0 +1,77 @@
+/obj/item/weapon/gun/energy/frame
+	name = "energy gun frame"
+	desc = "A frame for constructing an Energy Gun."
+	customizable_type = CUSTOMIZABLE_ENERGY
+	icon = 'icons/obj/guncrafting/main.dmi'
+	icon_state = "frame_energy"
+	charge_sections = 2
+	ammo_type = list()
+
+/obj/item/weapon/gun/energy/frame/testing
+	pin = null
+
+/obj/item/weapon/gun_attachment/frame/energy
+	hacky_as_fuck = /obj/item/weapon/gun/energy/frame
+
+
+/obj/item/weapon/gun/ballistic/frame
+	name = "projectile gun frame"
+	desc = "This is my rifle. There are many like it, but this one is mine. My rifle is my best friend. It is my life. I must master it as I must master my life. My rifle, without me, is useless. Without my rifle, I am useless. I must fire my rifle true. I must shoot straighter than my enemy who is trying to kill me. I must shoot him before he shoots me."
```

Comment:
please stop with this cutesy noise. it's confusing enough for players to transition to a new gun paradigm without having to wade through all the pop culture refs from your high school experience